src/indexmap.rs

```diff
@@ -209,12 +209,9 @@ where
                     // robin hood: steal the spot if it's better for us
                     let index = self.entries.len();
                     unsafe { self.entries.push_unchecked(Bucket { hash, key, value }) };
+                    Self::insert_phase_2(&mut self.indices, probe, Pos::new(index, hash));
```

Comment:
as an aside: do we need this function to return anything? it looks like the return value wasn't used anywhere besides this function, and it's not clear from the documentation what the intention for the return value is. for this case it's returning the last `probe` value while we are trying to use it as an entries index